pkg/kapp/clusterapply/applying_changes.go

```diff
@@ -88,22 +91,26 @@ func (c *ApplyingChanges) Apply(allChanges []*ctldgraph.Change) ([]WaitingChange
 
 			if result.Err != nil {
 				lastErr = result.Err
-				if result.Retryable {
-					continue
+				if !result.Retryable {
+					if c.exitOnError {
+						return nil, nil, result.Err
+					}
+					unsuccessfulChangeDesc = append(unsuccessfulChangeDesc, result.Err.Error())
```

Comment:
A thought in  line with the default behaviour, if we do decide to apply as many changes as we can by default.
Should we print a message like `ok: reconcile deployment/kapp-controller (apps/v1) namespace: kc` the moment a resource fails?

Not convinced about this myself. It does not benefit cases where kapp is being run as a part of a script.
However I am guessing I would not like to wait till everything has finished applying know that a resource has failed 🤔

Reply:
> Should we print a message like ok: reconcile deployment/kapp-controller (apps/v1) namespace: kc the moment a resource fails?

I didn't understand this part, do you mean we should print `ok: reconcile...` for a resource when it fails?

Reply:
Not sure if we should do this, but I think it might be a nice to have if the default behaviour is to apply maximally

Reply:
Ah, I see.

So, we are doing something like this
```bash
3:38:16PM: fail: reconcile job/my-job-fail (batch/v1) namespace: default
3:38:16PM:  ^ Failed with reason BackoffLimitExceeded: Job has reached the specified backoff limit
```
(^ Taken from the gist example)

Reply:
Oops! I missed that. Resolving this 🚀